vault/generate_root_recovery.go

```diff
@@ -19,6 +19,25 @@ type generateRecoveryToken struct {
 	token *atomic.String
 }
 
+func (g *generateRecoveryToken) authenticate(ctx context.Context, c *Core, combinedKey []byte) error {
+	key, err := c.unsealKeyToMasterKey(ctx, combinedKey)
```

Comment:
If an old style shamir is in use wouldn't this return an error since `VerifyMaster` requires Vault to already be unsealed?

Reply:
Ooh, good point.